packages/wdio-cucumber-framework/src/index.js

```diff
@@ -0,0 +1,201 @@
+        this.cid = cid
+        this.specs = specs
+        this.reporter = reporter
+        this.capabilities = capabilities
+        this.config = config
+        this.cucumberOpts = Object.assign(DEFAULT_OPTS, config.cucumberOpts)
+    }
+
+    async run () {
+        let runtimeError
+        let result
+
+        Cucumber.supportCodeLibraryBuilder.reset(this.cwd)
+
+        try {
+            this.registerCompilers()
+            this.loadSpecFiles()
+            this.wrapSteps()
+            Cucumber.setDefaultTimeout(this.cucumberOpts.timeout)
+            const supportCodeLibrary = Cucumber.supportCodeLibraryBuilder.finalize()
+
+            const eventBroadcaster = new EventEmitter()
+            // eslint-disable-next-line no-new
+            new Hookrunner(eventBroadcaster, this.config)
+            const reporterOptions = {
+                capabilities: this.capabilities,
+                ignoreUndefinedDefinitions: Boolean(this.cucumberOpts.ignoreUndefinedDefinitions),
+                failAmbiguousDefinitions: Boolean(this.cucumberOpts.failAmbiguousDefinitions),
+                tagsInTitle: Boolean(this.cucumberOpts.tagsInTitle)
+            }
+
+            this.cucumberReporter = new CucumberReporter(eventBroadcaster, reporterOptions, this.cid, this.specs, this.reporter)
+
+            const pickleFilter = new Cucumber.PickleFilter({
+                featurePaths: this.specs,
+                names: this.cucumberOpts.name,
+                tagExpression: this.cucumberOpts.tagExpression
+            })
+            const testCases = await Cucumber.getTestCasesFromFilesystem({
+                cwd: this.cwd,
+                eventBroadcaster,
+                featurePaths: this.specs,
+                order: this.cucumberOpts.order,
+                pickleFilter
+            })
+            const runtime = new Cucumber.Runtime({
+                eventBroadcaster,
+                options: this.cucumberOpts,
+                supportCodeLibrary,
+                testCases
+            })
+
+            await executeHooksWithArgs(this.config.before, [this.capabilities, this.specs])
+            result = await runtime.start() ? 0 : 1
+
+            /**
+             * if we ignore undefined definitions we trust the reporter
+             * with the fail count
+             */
+            if (this.cucumberOpts.ignoreUndefinedDefinitions && result) {
+                result = this.cucumberReporter.failedCount
+            }
+        } catch (e) {
+            runtimeError = e
+            result = 1
+        }
+
+        await executeHooksWithArgs(this.config.after, [runtimeError || result, this.capabilities, this.specs])
+
+        /**
+         * in case the spec has a runtime error throw after the wdio hook
+         */
+        if (runtimeError) {
+            throw runtimeError
+        }
+
+        return result
+    }
+
+    registerCompilers () {
+        this.cucumberOpts.compiler.forEach(compiler => {
```

Comment:
`compiler` has beed deprecated in favor of `require-module` https://github.com/cucumber/cucumber-js/blob/master/docs/cli.md#transpilation
NOTE: `require` and `require-module` are different things!

In addition, what if we have not passed any compilers/module? I think it should be allowed

Reply:
@mgrybyk would you mind doing this as a secondary PR? I would love for people to get the ability to update.